src/main/java/io/vavr/concurrent/FutureImpl.java

```diff
@@ -376,6 +376,9 @@ boolean tryComplete(Try<? extends T> value) {
             if (waiters != null) {
                 waiters.forEach(this::unlock);
             }
+            if(isCancelled()) {
```

Comment:
(I don't want to be a nitpicker but you missed one space `if(is...` --> `if (is...`)